cluster/src/main/java/org/apache/iotdb/cluster/query/ClusterPlanExecutor.java

```diff
@@ -373,8 +373,85 @@ protected void loadConfiguration(LoadConfigurationPlan plan) throws QueryProcess
         break;
       default:
         throw new QueryProcessException(String
-            .format("Unrecognized load configuration plan type: %s",
-                plan.getLoadConfigurationPlanType()));
+                .format("Unrecognized load configuration plan type: %s",
+                        plan.getLoadConfigurationPlanType()));
+    }
+  }
+
+  @Override
+  protected Pair<Set<String>, String> deleteTimeseries(String path) throws MetadataException {
+    ConcurrentHashMap<String, Set<String>> resultPair = new ConcurrentHashMap<>();
+    Pair<Set<String>, String> localPair = MManager.getInstance().deleteTimeseries(path);
+    resultPair.put(localPair.right, localPair.left);
+
+    ExecutorService pool = new ScheduledThreadPoolExecutor(THREAD_POOL_SIZE);
+
+    for (PartitionGroup group : metaGroupMember.getPartitionTable().getGlobalGroups()) {
+      Node header = group.getHeader();
+      if (header.equals(metaGroupMember.getThisNode())) {
+        continue;
+      }
+      pool.submit(() -> {
+        DeleteTimeseriesHandler handler = new DeleteTimeseriesHandler();
+        AtomicReference<DeleteTimeseriesRespPair> response = new AtomicReference<>(null);
+        handler.setResponse(response);
+
+        for (Node node : group) {
```

Comment:
As a node exists in multiple groups, a node will perform the same deletion several times and I am not sure if it is meaningful. Why not just send the deletion to each node in allNodes?

Reply:
Thanks for your advice. I've changed the logic.